### ISSUE TITLE
Most tests are failing in my environment

### ISSUE THREAD

**Issue by MeggyCal** [encode/httpcore#108]
Hi,

I am trying to build this package and I am constantly getting errors. I am wondering why. Could you please take a look? The package lives at https://build.opensuse.org/package/show/home:mcalabkova:branches:devel:languages:python/python-httpcore.

In short: the software constantly refuses to use HTTP2 even when it is allowed (first half of errors). HTTP1.1 tests then fail on the return code (second half of errors). Do you have any idea what could cause this?

Known problems: our buildservice doesn't have access to internet during the build. Also we do not have all the packages you are pulling in during your tests, but I think I have everything mentioned in the [test] subsection of requirements.txt.

Thanks a lot!

**Comment by yeraydiazdiaz** [comment]
Hi @MeggyCal, our test suite performs real internet connections, you can see in the logs the DNS fails and eventually the connection times out.

We don't strictly have to, in fact HTTPX uses a local HTTP server, but real internet connections give us complete confidence and it's worked well for us up until now, we might revisit that decision in the future though, as we increase our test coverage.

I also noticed there are some jobs that fail because they're trying to install HTTP Core on Python 2.7 which is unsupported.

In short, I don't think we'll be changing the nature of your test failures anytime soon, but I'll open an issue to discuss moving to local test server.
